controllers/nnf_storage_controller.go

```diff
@@ -349,67 +328,29 @@ func (r *NnfStorageReconciler) aggregateNodeStorageStatus(ctx context.Context, s
 // that aren't in the cache and we may not have been able to add the object reference
 // to the NnfStorage.
 func (r *NnfStorageReconciler) teardownStorage(ctx context.Context, statusUpdater *storageStatusUpdater, storage *nnfv1alpha1.NnfStorage) (nodeStoragesState, error) {
-	log := r.Log.WithValues("Workflow", types.NamespacedName{Name: storage.Name, Namespace: storage.Namespace})
-	var firstErr error
-
 	// Collect status information from the NnfNodeStorage resources and aggregate it into the
 	// NnfStorage
-	for i := range storage.Spec.AllocationSets {
+	for i := range storage.Status.AllocationSets {
 		_, err := r.aggregateNodeStorageStatus(ctx, statusUpdater, storage, i)
 		if err != nil {
 			return nodeStoragesExist, err
 		}
 	}
 
-	state := nodeStoragesDeleted
-
-	for allocationSetIndex, allocationSet := range storage.Spec.AllocationSets {
-		for i, node := range allocationSet.Nodes {
-			namespacedName := types.NamespacedName{
-				Name:      nnfNodeStorageName(storage, allocationSetIndex, i),
-				Namespace: node.Name,
-			}
-
-			// Do a Get on the nnfNodeStorage first to check if it's already been marked
-			// for deletion.
-			nnfNodeStorage := &nnfv1alpha1.NnfNodeStorage{}
-			err := r.Get(ctx, namespacedName, nnfNodeStorage)
-			if err != nil {
-				if !apierrors.IsNotFound(err) {
-					if firstErr == nil {
-						firstErr = err
-					}
-					log.Info("Unable to get NnfNodeStorage", "Error", err, "NnfNodeStorage", namespacedName)
-				}
-			} else {
-				state = nodeStoragesExist
-				if !nnfNodeStorage.GetDeletionTimestamp().IsZero() {
-					continue
-				}
-			}
+	childObjects := []dwsv1alpha1.ObjectList{
+		&nnfv1alpha1.NnfNodeStorageList{},
```

Comment:
Add NnfStorageProfile here.  Add your labels to it as well, because it's owned by the NnfStorage.
Look for setPinnedProfileOwnership().

Or are there certain kinds of resources that don't require your new managed deletion?

Reply:
Nope it should have it too. I just missed it.

Reply:
Can we pull these lists out and put them, empty, in `api/*/foo_types.go`?  Then populate the lists in each  SetupWithManager().  I'd like to get the lists to a place where they're easy to find and where future devs are more likely to maintain them as new CRDs are created.  Building them in SetupWithManager() puts them in a noticeable place as people are updating their Owns() and Watches().